bug: Broken pipeline after updating to kubebuilder v4

## Description
Pipeline does not work anymore.

## How to Reproduce
Run the pipeline after changes, cause would probably be updating the golang version in the pipeline. This could potentially be fixed by rolling back. The cause could also be related to the structure change.

## Logs/Error Message
```
vet: cmd/main.go:83:3: unknown field MetricsBindAddress in struct literal of type manager.Options
```

```
no required module provides package github.com/holy-tech/Mission-Control-Operator; to add it:
        go get github.com/holy-tech/Mission-Control-Operator
```

```
ERRO Running error: context loading failed: failed to load packages: failed to load with go/packages: err: exit status 1: stderr: go: errors parsing go.mod:
/root/project/go.mod:3: invalid go version '1.21.0': must match format 1.23 
make: *** [Makefile:171: ci-lint] Error 3
```
## Screenshots

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -7,7 +7,7 @@ jobs:
     working_directory: /go/src/Mission-Control-Operator
     steps:
       - checkout
-      - run: go install github.com/holy-tech/Mission-Control-Operator
+      - run: go build -v ./...
   lint:
     docker:
       - image: golangci/golangci-lint:v1.54.1

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -10,7 +10,7 @@ jobs:
       - run: go install github.com/holy-tech/Mission-Control-Operator
   lint:
     docker:
-      - image: golangci/golangci-lint:v1.53
+      - image: golangci/golangci-lint:v1.54.1
     steps:
       - checkout
       - run: make ci-lint | tee golanci-report.xml

